Functions/InModuleScope.Tests.ps1

```diff
@@ -18,27 +18,70 @@ Describe "Module scope separation" {
     }
 }
 
-Describe "Executing test code inside a module" {
+Describe "Executing test code inside a 'Script' module" {
+
     New-Module -Name TestModule {
         function InternalFunction { 'I am the internal function' }
         function PublicFunction   { InternalFunction }
         Export-ModuleMember -Function PublicFunction
     } | Import-Module -Force
 
-    It "Cannot call module internal functions, by default" {
+    It "Ensure we've created a 'Script' module" {
+        $module = Get-Module -Name "TestModule";
+        $module.ModuleType | Should Be "Script";
+    }
+
+    It "Cannot call module internal functions from a 'Script' module, by default" {
         { InternalFunction } | Should Throw
     }
 
     InModuleScope TestModule {
-        It "Can call module internal functions using InModuleScope" {
+
+        It "Can call module internal functions from a 'Script' module using InModuleScope" {
             InternalFunction | Should Be 'I am the internal function'
         }
 
-        It "Can mock functions inside the module without using Mock -ModuleName" {
+        It "Can mock functions inside a 'Script' module without using Mock -ModuleName" {
             Mock InternalFunction { 'I am the mock function.' }
             InternalFunction | Should Be 'I am the mock function.'
         }
     }
 
     Remove-Module TestModule -Force
+
+}
+
+Describe "Executing test code inside a 'Manifest' module" {
+
+    New-Item -Type "Directory" -Path "TestDrive:\MyModule";
+
+    Set-Content -Path "TestDrive:\MyModule\Invoke-PublicMethod.ps1"  -Value 'function Invoke-PublicMethod { return Invoke-PrivateMethod; }';
```

Comment:
We adjust Pester to work on PowerShell Core (Windows/Linux/macOS) - please check the pull request #925.

Paths used by you will not be compatible with non-Windows systems :-/

Please check how I did it [here](https://github.com/pester/Pester/pull/925/commits/d999adfb2860aa781233df673e899457ad6b736b).